src/components/DocumentModals/ShowDocumentsModal.jsx

```diff
@@ -0,0 +1,131 @@
+// React Imports
+import React, { useState } from 'react';
+// Inrupt Imports
+import { useSession } from '@inrupt/solid-ui-react';
+// Material UI Imports
+import Box from '@mui/material/Box';
+import Button from '@mui/material/Button';
+import Dialog from '@mui/material/Dialog';
+import DialogActions from '@mui/material/DialogActions';
+import DialogContent from '@mui/material/DialogContent';
+import DialogTitle from '@mui/material/DialogTitle';
+import ClearIcon from '@mui/icons-material/Clear';
+import Typography from '@mui/material/Typography';
+// Other Library Imports
+import { v4 as uuidv4 } from 'uuid';
+// Utility Imports
+import { getBlobFromSolid } from '../../utils';
+import ShowDocumentLocalModal from './ShowDocumentLocalModal';
+
+/**
+ * ShowDocumentsModal Component - Popup modal that shows client documents if gotten
+ * explicit permission via ACL
+ *
+ * @memberof DocumentModals
+ * @name ShowDocumentsModal
+ */
+
+const ShowDocumentsModal = ({ showModal, setShowModal, fileSrc }) => {
+  const { session } = useSession();
+
+  const [showDocumentBlob, setShowDocumentBlob] = useState(false);
+  const [fileBlobUrl, setFileBlobUrl] = useState('');
+
+  const handleShowDocumentLocal = async (fileUrl) => {
+    const urlFileBlob = await getBlobFromSolid(session, fileUrl);
+    setFileBlobUrl(urlFileBlob);
+    setShowDocumentBlob(true);
+  };
+
+  return (
+    <Dialog
+      fullScreen
+      open={showModal}
+      aria-labelledby="dialog-title"
+      onClose={() => setShowModal(false)}
+    >
+      <DialogTitle id="dialog-tile">Documents</DialogTitle>
+
+      <DialogContent>
+        <Box
+          sx={{
+            display: 'flex',
+            flexWrap: 'wrap',
+            alignItems: 'center',
+            justifyContent: 'center',
+            gap: 5
+          }}
+        >
+          {fileSrc.map((src) => (
+            <Box
+              key={uuidv4()}
```

Comment:
what is this?  I've been meaning to ask for a while what this library was

Reply:
It's a function for generating unique keys if you do not have one but need something.

Reply:
It would be better to use file names or some other value on the `src` as the keys. Then the key can be used by screen readers or other parts of the code base, or used in testing. As is, this will generate a meaningless number every single time the code is run.

Reply:
Think that would be good. Could make sure it's fully unique with the url path to the file.

I'll give that a go in place of the uuid.